File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        // throw new NotImplementedException("Please implement the (static) LogLine.Message() method");
        int index = logLine.IndexOf(": ");       
        string message = logLine.Substring(index + 2);
        message = message.Replace("\t", "").Replace("\r", "").Replace("\n", "").Trim();        
        return message;
    }

    public static string LogLevel(string logLine)
    {
        // throw new NotImplementedException("Please implement the (static) LogLine.LogLevel() method");
        int index = logLine.IndexOf(": ");       
        string message = logLine.Remove(index);
        message = message.Trim('[',']');
        message = message.ToLower();
        return message;
    }

    public static string Reformat(string logLine)
    {
        // throw new NotImplementedException("Please implement the (static) LogLine.Reformat() method");
        int index = logLine.IndexOf(": ");       
        string level = logLine.Remove(index);
        level = level.Replace("[","(").Replace("]",")");
        level = level.ToLower();
               
        string message = logLine.Substring(index + 2);
        message = message.Replace("\t", "").Replace("\r", "").Replace("\n", "").Trim();

        string answer = message + " " + level;
        return answer;
        
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>
#define EXERCISM_RUN_ALL_TESTS

namespace log_line {

    // Magic numbers and no error handling. eep
std::string message(std::string line) {
    //  [Error]: string
    auto message = line.substr( line.find(':') + 2, line.length() );
    return message;
}

std::string log_level(std::string line) {

auto level = line.substr( 1, line.find(']') - 1);
    return level;
}

std::string reformat(std::string line) {
    auto message = log_line::message(line);
    auto level = log_line::log_level(line);

std::string reformedString = message  + ' ' + '(' + level + ')';

    return reformedString;
}
}  // namespace log_line
